### ISSUE TITLE
Error when trying to edit mobile notifications in android app

### ISSUE THREAD

**Issue by bj0rj0h** [mattermost/mattermost-mobile#1014]
Submit feature requests to http://www.mattermost.org/feature-requests/. File non-security related bugs here in the following format: 

#### Summary
APP crashed when trying to edit notification settings on mobile. 

#### Environment Information

- Device Name: OnePlus3t
- OS Version: 7.1.1
- Mattermost App Version: 1.3.0 (Build 57)
- Mattermost Server Version: 4.2.0 (Build 4.2.0)

#### Steps to reproduce
Entered settings to modify notification settings for mobile notifications. 

Settings-->notifications-->mobile
When I clicked "mobile" it crashed

#### Expected behavior
Show me settings for mobile notifications

#### Observed behavior
When I clicked mobile the app gave me following screen

![screenshot_20171012](https://user-images.githubusercontent.com/9116059/31511690-cdf9d87a-af88-11e7-9a18-fbd123cb3e07.jpg)

#### Possible fixes
[stack-trace.txt](https://github.com/mattermost/mattermost-mobile/files/1380233/stack-trace.txt)

**Comment by mthornba** [comment]
I'm seeing the identical behaviour plus an additional error.

**Summary**

APP crashed when trying to edit notification settings on mobile.

**Environment Information**

Device Name: Galaxy S6
OS Version: 7.0
Mattermost App Version: 1.3.0 (Build 57)
Mattermost Server Version: 4.3.0

**Steps to reproduce**

Entered settings to modify notification settings for mobile notifications.

Settings-->notifications-->mobile
When I clicked "mobile" it crashed

**Expected behavior**

Show me settings for mobile notifications

**Observed behavior**

When I clicked mobile the app gave me the same screen as in @bj0rj0h's post above.
Clicking the "relaunch" button clears the error dialog but does not appear to restart the app.  The app remains open on the _Notifications_ screen. At this point, clicking the _Mobile_ button again results in the following screen:
![image](https://user-images.githubusercontent.com/13801379/31682643-b7176dd4-b348-11e7-8b43-609b4911b91f.png)

Additional data has been uploaded here:
https://mattermost.zendesk.com/hc/en-us/requests/5006#
using the _Report a Problem_ link in the Android app menu.
